flink-runtime/src/main/java/org/apache/flink/runtime/deployment/InputGateDeploymentDescriptor.java

```diff
@@ -115,7 +118,18 @@ public ResultPartitionType getConsumedPartitionType() {
 
     @Nonnegative
     public int getConsumedSubpartitionIndex() {
-        return consumedSubpartitionIndex;
+        checkState(
+                consumedSubpartitionIndexRange.startIndex
+                        == consumedSubpartitionIndexRange.endIndex);
+        return consumedSubpartitionIndexRange.startIndex;
+    }
+
+    /**
+     * Return the index range of the the consumed subpartitions. This method will be used by
```

Comment:
> This method will be used by adaptive scheduler.

Let's remove this comment because it's not very accurate. b.t.w. adaptive scheduler current mainly refers to the adaptive streaming scheduler.